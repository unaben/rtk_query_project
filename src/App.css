h1 {
  text-align: center;
}

.data {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
.data > span:first-child {
  font-weight: 500;
}
