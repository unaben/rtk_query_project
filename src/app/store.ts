import { configureStore } from "@reduxjs/toolkit";
import { productsApi } from "../services/ProductsApi";

export const store = configureStore({
  reducer: {
    // Add the generated reducer as a specific top-level slice
    [productsApi.reducerPath]: productsApi.reducer,
  },
  // Adding the api middleware enables caching, invalidation, polling,
  // and other useful features of `rtk-query`.
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(productsApi.middleware),
});
