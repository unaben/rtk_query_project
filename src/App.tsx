import { FC } from "react";
import "./App.css";
import { MultiActionFunc } from "./components/MultiActionFunc";
import ProductDetail from "./components/ProductDetail";
import { Product } from "./model/product";
import { useGetAllProductsQuery } from "./services/ProductsApi";

const App: FC = () => {
  const { data : products, error, isLoading, isFetching, isSuccess } =
    useGetAllProductsQuery();
  console.log(products);
  return (
    <div className="App">
      <h1>RTK Query</h1>
      {isLoading && <h2>...Loading</h2>}
      {isFetching && <h2>...isFetching</h2>}
      {error && <h2>Something went wrong</h2>}
      {isSuccess && (
        <div>
          {products?.map((product: Product) => {
            return (
              <div className="data" key={product.id}>
                <span>{product.name}</span>
                <span>
                  <ProductDetail id={product.id} />
                </span>
              </div>
            );
          })}
          <MultiActionFunc />
          <div></div>
        </div>
      )}
    </div>
  );
};

export default App;
