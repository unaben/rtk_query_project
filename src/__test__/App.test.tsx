import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../app/store";
import App from "../App";

describe("App component", () => {
  it("should renders correctly", () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const text = screen.getByText(/RTK Query/i);
    expect(text).toBeInTheDocument();
  });

  it("should match snapshot", async () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
