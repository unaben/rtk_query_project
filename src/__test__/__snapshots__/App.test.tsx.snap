// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      RTK Query
    </h1>
    <h2>
      ...Loading
    </h2>
    <h2>
      ...isFetching
    </h2>
  </div>
</div>
`;
